<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CTimeoutsDialog</class>
 <widget class="QDialog" name="CTimeoutsDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>464</width>
    <height>323</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Timeouts configuration</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QLabel" name="label">
           <property name="text">
            <string>&amp;Max error count before connection stop</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="buddy">
            <cstring>spinMaxRecErrorCount</cstring>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="spinMaxRecErrorCount">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Maximum error count for daveReadBytes function. &lt;/p&gt;&lt;p&gt;This counter increases in each PLC cycle on connection failures, inaccessible or incorrect addresses of PLC memory areas, specified in variables.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="maximum">
            <number>99999</number>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QLabel" name="label_4">
           <property name="text">
            <string>(0 - don't stop)</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>TCP &amp;connect retries when starting connection</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="buddy">
            <cstring>spinConnectRetryCount</cstring>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QSpinBox" name="spinConnectRetryCount">
           <property name="toolTip">
            <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;TCP connection retries when starting online mode.&lt;/p&gt;&lt;p&gt;Do not set &lt;span style=&quot; font-weight:600;&quot;&gt;this parameter&lt;/span&gt; and &lt;span style=&quot; font-weight:600;&quot;&gt;reconnection retries count&lt;/span&gt; both equal to zero.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
           </property>
           <property name="maximum">
            <number>99999</number>
           </property>
          </widget>
         </item>
         <item row="1" column="2">
          <widget class="QLabel" name="label_5">
           <property name="text">
            <string>(0 - infinite)</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_3">
           <property name="text">
            <string>Network timeout / &amp;TCP (ms)</string>
           </property>
           <property name="buddy">
            <cstring>spinNetTimeout</cstring>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QSpinBox" name="spinNetTimeout">
           <property name="minimum">
            <number>1000</number>
           </property>
           <property name="maximum">
            <number>120000</number>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QLabel" name="label_8">
           <property name="text">
            <string>&amp;Pause before connection retry (sec)</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="buddy">
            <cstring>spinWaitBeforeRetryConnect</cstring>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QSpinBox" name="spinWaitBeforeRetryConnect">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>99999</number>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="label_6">
           <property name="text">
            <string>&amp;Failed connection retry count</string>
           </property>
           <property name="wordWrap">
            <bool>true</bool>
           </property>
           <property name="buddy">
            <cstring>spinTotalRetryCount</cstring>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSpinBox" name="spinTotalRetryCount"/>
         </item>
         <item row="4" column="2">
          <widget class="QLabel" name="label_7">
           <property name="text">
            <string>(0 - infinite)</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QCheckBox" name="checkSuppressMsgBoxes">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Suppress message boxes with error and warning messages. All messages saving to log window.&lt;/p&gt;&lt;p&gt;Very useful and memory conservative on long background works when starting with '-start' key from command prompt.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Do not show interactive message boxes</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="checkRestoreCSV">
         <property name="toolTip">
          <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;This settings only for interactive mode.&lt;/p&gt;&lt;p&gt;When starting with '-start' key from command promnt, this option is always activated.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
         </property>
         <property name="text">
          <string>Restore CSV recording after connection failure</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>OK</string>
       </property>
       <property name="default">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>Cancel</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>spinMaxRecErrorCount</tabstop>
  <tabstop>spinConnectRetryCount</tabstop>
  <tabstop>spinNetTimeout</tabstop>
  <tabstop>spinWaitBeforeRetryConnect</tabstop>
  <tabstop>spinTotalRetryCount</tabstop>
  <tabstop>checkSuppressMsgBoxes</tabstop>
  <tabstop>checkRestoreCSV</tabstop>
  <tabstop>pushButton</tabstop>
  <tabstop>pushButton_2</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>CTimeoutsDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>CTimeoutsDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>219</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>23</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
